docs/dev/serialization.md

```diff
@@ -0,0 +1,110 @@
+This developer document explains how Cirq serializes objects into (and out of) JSON.
+It also explains how to add a new serializable object,
+how to remove a serializable object from cirq while maintaing backwards compatibility with old serialized files,
+and various related guidelines.
+
+## Exposed API
```

Comment:
Optional: should there be a short background doc to explain why we want/need this?

Reply:
Added.